File: backend/app/schemas/procurement.py
from pydantic import BaseModel, ConfigDict, Field
from typing import Optional
from datetime import datetime

class ProcurementBase(BaseModel):
    item_name: str = Field(..., min_length=1)
    project_id: str
    quantity: float = Field(1.0, ge=0)
    unit: Optional[str] = None
    estimated_cost: Optional[float] = Field(None, ge=0)
    actual_cost: Optional[float] = Field(None, ge=0)
    status: Optional[str] = "PLANNED"
    notes: Optional[str] = None

class ProcurementCreate(ProcurementBase):
    pass

class ProcurementUpdate(BaseModel):
    item_name: Optional[str] = None
    quantity: Optional[float] = None
    unit: Optional[str] = None
    estimated_cost: Optional[float] = None
    actual_cost: Optional[float] = None
    status: Optional[str] = None
    notes: Optional[str] = None

class ProcurementResponse(ProcurementBase):
    id: str
    created_at: datetime
    updated_at: Optional[datetime] = None

    model_config = ConfigDict(from_attributes=True)

File: frontend/src/types/procurement.ts
// RFQ Types
export type RFQStatus = 'DRAFT' | 'ISSUED' | 'EVALUATING' | 'AWARDED' | 'CLOSED';

export interface RFQItem {
  id: string;
  rfq_number: string;
  title: string;
  project_id: string;
  budget: number;
  status: RFQStatus;
  due_date: string;
  description?: string;
  created_at?: string;
  updated_at?: string;
}

export interface RFQCreateInput {
  rfq_number: string;
  title: string;
  project_id: string;
  budget: number;
  status: RFQStatus;
  due_date: string;
  description?: string;
}

// Purchase Order Types
export type POStatus = 'DRAFT' | 'ISSUED' | 'IN_TRANSIT' | 'FULFILLED' | 'CANCELLED';

export interface PurchaseOrderItem {
  id: string;
  po_number: string;
  project_id: string;
  vendor_name: string;
  total_amount: number;
  status: POStatus;
  issued_date: string;
  delivery_date?: string;
  notes?: string;
  created_at?: string;
  updated_at?: string;
}

export interface POCreateInput {
  po_number: string;
  project_id: string;
  vendor_name: string;
  total_amount: number;
  status: POStatus;
  issued_date: string;
  delivery_date?: string;
  notes?: string;
}

// Procurement Item Types
export type ProcurementStatus = 'PENDING' | 'ORDERED' | 'SHIPPED' | 'DELIVERED' | 'CANCELLED';

export interface ProcurementLineItem {
  id: string;
  item_name: string;
  project_id: string;
  quantity: number;
  unit: string;
  estimated_cost: number;
  actual_cost?: number;
  status: ProcurementStatus;
  notes?: string;
  created_at?: string;
  updated_at?: string;
}

export interface ProcurementItemCreateInput {
  item_name: string;
  project_id: string;
  quantity: number;
  unit: string;
  estimated_cost: number;
  actual_cost?: number;
  status: ProcurementStatus;
  notes?: string;
}
